Automattic/pocket-casts-android
# Implement Small Player widget

## Description

This targets a separate branch instead of `main` because widgets cannot be feature flagged as they are compile time dependent. Once all features are implement I will make a PR for the `task/widgets` branch.

Designs: Lkt7fuf9Nq3XvfAFPCfpTD-fi-1044_7170

There are some known issues:

1. Sometimes artwork doesn't load and placeholder is displayed instead. Even though the `Image` composable receives a correct `ImageProvider` it still doesn't update. It might be a bug withing the `Glance` library. However image loads correctly once the playback is resumed.
2. Widget's preview and placeholder are currently not implemented. Internal ref: p1711476556931619/1711396714.667229-slack-C05RR9P9RAT
3. Playback button doesn't scale with image. There is no API for that in the `Glance` library.

## Testing Instructions

1. Test that app opens when nothing is in the queue and you tap on the widget.
4. Test that the playback start and stops when you tap on the on the widget while episodes are in queue.
5. Test that the widget respects `Use embedded RSS artwork`.
6. Test that the widget respects `Use dynamic colors for widget` and Dark/Light themes. You can verify it with the playback button color.
7. Test that the widget updates artwork when currently playing episode is changed.

## Screenshots or Screencast 

### Grid sizes

| 5x5 | 4x4 | 3x3 | 2x2 |
| - | - | - | - |
| ![5x5](https://github.com/Automattic/pocket-casts-android/assets/30936061/30b847ad-6615-4504-9ca6-e0249b75f458) | ![4x4](https://github.com/Automattic/pocket-casts-android/assets/30936061/92b64913-0a03-49ec-8373-912ac8c3af95) | ![3x3](https://github.com/Automattic/pocket-casts-android/assets/30936061/5d3c6dcc-ad53-4e94-bc5b-89b312adb31b) | ![2x2](https://github.com/Automattic/pocket-casts-android/assets/30936061/1a8feeb4-20ca-4ede-b878-5374147d408c) |

### Artwork

| Empty | Podcast | Episode |
| - | - | - |
| ![empty](https://github.com/Automattic/pocket-casts-android/assets/30936061/310954e9-8498-4877-a808-b91e7c29af50) | ![podcast](https://github.com/Automattic/pocket-casts-android/assets/30936061/89e2c5af-e816-4d3b-9d68-2380ce2b41ac) | ![episode](https://github.com/Automattic/pocket-casts-android/assets/30936061/75734d4f-49ec-4151-a799-db69eb08f36f)  |

### Dynamic theme

| A | B | C | D | E | F |
| - | - | - | - | - | - |
| ![Screenshot_20240327-105810](https://github.com/Automattic/pocket-casts-android/assets/30936061/fcafcf31-c1cf-4e94-8615-7b194c67ad8d) | ![Screenshot_20240327-105847](https://github.com/Automattic/pocket-casts-android/assets/30936061/42758648-e61c-4e6d-9af6-1df68b0cd64b) | ![Screenshot_20240327-105918](https://github.com/Automattic/pocket-casts-android/assets/30936061/ed0e7d99-0231-4ee3-a12c-e592881b304b) | ![Screenshot_20240327-110002](https://github.com/Automattic/pocket-casts-android/assets/30936061/64a1f569-6035-4f52-9397-eda3fcc79f1e) | ![Screenshot_20240327-110049](https://github.com/Automattic/pocket-casts-android/assets/30936061/dd299882-2c18-4aff-9799-5d521d127f51) | ![Screenshot_20240327-110122](https://github.com/Automattic/pocket-casts-android/assets/30936061/e7d26ba4-8ffe-421c-81ed-2fc25778e6e7) |

## Checklist
 
#### I have tested any UI changes...
